src/firebase-resource-manager.ts

```diff
@@ -0,0 +1,178 @@
+    logger.info("");
+    logger.info("Project information:");
+    logger.info(`   - Project ID: ${clc.bold(projectData.projectId)}`);
+    logger.info(`   - Project Name: ${clc.bold(projectData.displayName)}`);
+    logger.info("");
+    logger.info("Firebase console is available at");
+    logger.info(`https://console.firebase.google.com/project/${clc.bold(projectId)}/overview`);
+    return { projectNumber, projectId };
+  }
+
+  /**
+   * @return {Promise} this function returns a promise that resolves to the resource name of the
+   *     create cloud project LRO, or rejects if an error is thrown
+   */
+  private async createCloudProject(
+    projectId: string,
+    projectDisplayName: string,
+    parentResource?: ParentResource
+  ): Promise<string> {
+    const spinner = new OraWrapper("Creating Google Cloud Platform project");
+    spinner.start();
+
+    try {
+      const response = await api.request("POST", "/v1/projects", {
+        auth: true,
+        origin: api.resourceManagerOrigin,
+        timeout: 15 * ONE_SECOND_MILLIS,
+        data: { projectId, name: projectDisplayName, parent: parentResource },
+      });
+      spinner.succeed();
+      return response.body.name; /* Operation resource name */
+    } catch (err) {
+      spinner.fail();
+      logger.debug(err.message);
+      throw new FirebaseError(
+        "Failed to create Google Cloud project. See firebase-debug.log for more info.",
+        { exit: 2, original: err }
+      );
+    }
+  }
+
+  /**
+   * @return {Promise} this function returns a promise that resolves to the new cloud project
+   *     information
+   */
+  private async pollCloudProjectCreationOperation(operationResourceName: string): Promise<any> {
+    const pollerOptions: LroPollerOptions = {
+      pollerName: "Project Creation Poller",
+      apiOrigin: api.resourceManagerOrigin,
+      apiVersion: "v1",
+      operationResourceName,
+    };
+    const spinner = new OraWrapper("Waiting for project creation to be completed");
+    spinner.start();
+
+    try {
+      const { error, response } = await this.poller.poll(pollerOptions);
+      if (error) {
+        throw error;
+      }
+      spinner.succeed();
+      return response;
+    } catch (err) {
+      spinner.fail();
+      logger.debug(err.message);
+      throw new FirebaseError(
+        "Failed to create Google Cloud project. See firebase-debug.log for more info.",
+        { exit: 2, original: err }
+      );
+    }
+  }
+
+  /**
+   * @return {Promise} this function returns a promise that resolves to the resource name of the add
+   *     Firebase to cloud project LRO, or rejects if an error is thrown
+   */
+  private async addFirebaseToCloudProject(projectId: string): Promise<string> {
+    const spinner = new OraWrapper("Adding Firebase to Google Cloud project");
+    spinner.start();
+
+    // TODO(caot): Removed when "Deferred Analytics" and "Deferred Location" are launched
```

Comment:
Let's leave this comment open until this TODO is resolved, so that it's not forgotten.